vlasovsolver/cpu_trans_pencils.cpp

```diff
@@ -0,0 +1,1170 @@
+      iy = 1;
+      break;
+   }
+   //ix = (dimension + 1) % 3; // incorrect for DCCRG
+   //iy = (dimension + 2) % 3;
+
+   x = coordinates[ix];
+   y = coordinates[iy];
+
+   pencils.addPencil(ids,x,y,periodic,path);
+   return;
+}
+
+/* Determine which cells in the local DCCRG mesh should be starting points for pencils.
+ * If a neighbor cell is non-local, across a periodic boundary, or in non-periodic boundary layer 1
+ * then we use this cell as a seed for pencils
+ *
+ * @param [in] mpiGrid DCCRG grid object
+ * @param [in] localPropagatedCells List of local cells that get propagated
+ * ie. not L2-boundary or DO_NOT_COMPUTE
+ * @param [in] dimension Spatial dimension
+ * @param [out] seedIds list of cell ids that will be starting points for pencils
+ */
+void getSeedIds(const dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>& mpiGrid,
+                const vector<CellID> &localPropagatedCells,
+                const uint dimension,
+                vector<CellID> &seedIds) {
+
+   const bool debug = false;
+   int myRank;
+   if (debug) MPI_Comm_rank(MPI_COMM_WORLD,&myRank);
+
+   // These neighborhoods now include the AMR addition beyond the regular vlasov stencil
+   int neighborhood = getNeighborhood(dimension,VLASOV_STENCIL_WIDTH);
+
+#pragma omp parallel for
+   for (uint i=0; i<localPropagatedCells.size(); i++) {
+      CellID celli = localPropagatedCells[i];
+
+      bool addToSeedIds = P::amrTransShortPencils;
+      if (addToSeedIds) {
+#pragma omp critical
+         seedIds.push_back(celli);
+         continue;
+      }
+      auto myIndices = mpiGrid.mapping.get_indices(celli);
+      int myRefLevel;
+
+      /* -----------------------------------------
+         | A |   | B |   |_|_|_|_|   |   | C |   |
+         |   |   |   |   | | | | |   |   |   |   |
+         -----------------------------------------
+         For optimal pencil generation, we need seedids at A, B, and C.
+         A Is triggered in the first if-clause. Pencils starting from B
+         will be split (but won't cause A to split), and pencils from
+         C will be able to remain again un-split. These checks need to be done
+         only if we aren't already at the maximum refinement level.
+      */
+
+      // First check negative face neighbors (A)
+      // Returns all neighbors as (id, direction-dimension) pair pointers.
+      for (const auto& [neighbor, dir] : mpiGrid.get_face_neighbors_of(celli) ) {
+         if ( dir == -((int)dimension + 1) ) {
+            // Check that the neighbor is not across a periodic boundary by calculating
+            // the distance in indices between this cell and its neighbor.
+            auto nbrIndices = mpiGrid.mapping.get_indices(neighbor);
+
+            // If a neighbor is across a periodic boundary, non-local, or
+            // in non-periodic boundary layer 2
+            // then we use the current cell as a seed for pencils
+            if ( (myIndices[dimension] < nbrIndices[dimension]) ||
```

Comment:
So this point should fix the issue with a 2-cell wide periodic domain.

It used to read along the lines of:
```
	    if ( abs ( (int64_t)(myIndices[dimension] - nbrIndices[dimension]) ) >
		 pow(2,mpiGrid.get_maximum_refinement_level()) ||
```
but as in this section we are always looking at negative direction neighbours, I think a simple greater-than check should be enough. Thoughts? @ursg @ykempf

Reply:
So this checks if my index is larger than my left neighbour's, correct? And that means my left neighbour is across the periodic wall?

Reply:
Well, smaller than, but yes. Left neighbours usually have a smaller index, so a larger index would mean it's actually further along in the grid, which realistically should mean it's across a periodic boundary.

Reply:
smaller, of course – I agree with this then.